lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -191,10 +191,23 @@ def find_or_create_for(application, resource_owner_id, scopes, expires_in, use_r
       # @return [Doorkeeper::AccessToken] array of matching AccessToken objects
       #
       def authorized_tokens_for(application_id, resource_owner_id)
-        ordered_by(:created_at, :desc)
+        # In case of static token expiry times, we can optimize fetching tokens
+        # by already filtering out tokens whose TTL has expired.
+        # We can't do that if custom expirations times have been configured, though.
+        ttl = unless Doorkeeper.configuration.option_defined?(:custom_access_token_expires_in)
+                Doorkeeper.configuration.access_token_expires_in.seconds.ago
+              end
+
+        tokens = ordered_by(:created_at, :desc)
           .where(application_id: application_id,
                  resource_owner_id: resource_owner_id,
                  revoked_at: nil)
+
+        if ttl
+          tokens.where("created_at >= ?", ttl)
```

Comment:
Can we move it to the scope? We need it for any other ORM extension to make integrration as easy as possible. So mongoid and sequel extensions can just introduce their own scopes instead of reimplementing this method fully.

Reply:
I now added a scope `Doorkeeper::AccessToken.created_since(time)`, I hope that was what you were asking for ^^